### PR TITLE
Fix AIM duping items when pulling out of a veh

### LINKED ISSUE #11999: Advanced Inventory item duplication
The way this could be replicated is by having a tab set to [A]ll and moving items from a vehicle (source is VH). If the specific tile is selected instead of [A]ll then it behaves normally.
The other tab doesn't seem to matter.

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/11009170-advanced-inventory-item-duplication?utm_campaign=plugin&utm_content=tracker%2F146201&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F146201&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by Coolthulhu**
Closes #11999

### CODE REVIEW

File: src/advanced_inv.cpp [hunk 1]
```diff
@@ -1752,15 +1753,18 @@ bool advanced_inventory::remove_item( advanced_inv_listitem &sitem )
...
+    }
+
+    if( !rc ) {
         g->m.i_rem( s.pos, sitem.it );
-        rc = true;
+        rc = true; // NOT ALWAYS! FIX!
     }
-    return rc;
+
+    return rc; // Always returns true
```
Comment (BevapDin) [inline src/advanced_inv.cpp]: > Always returns true

Than why make it a variable? Variable comes from good old Latin variare which apparently means “to change”, but if it's always the same, there is no change (-:

That said, the function should not return anything. It's already to late to react! The caller *must* make sure that removing the item is possible. Coincidentally, `map::i_rem` does not return anything either.

Comment (Coolthulhu) [inline src/advanced_inv.cpp]: I'm marking it as a "FIX THIS CRAP LATER, IT'S UGLY" quick fix hack for now, later on the `map::i_rem` will probably be changed to return bool, like `vehicle::remove_item`.

It is intentionally "wrong".

Comment (BevapDin) [inline src/advanced_inv.cpp]: Why do you want those function to return a value? What are you going to do with it? Besides displaying a debug message, which can happen already be done in the specific remove function.

Comment (Coolthulhu) [inline src/advanced_inv.cpp]: I wasn't sure, didn't really analyze the code there (just wanted a quick fix for a quick merge to fix some semi-old bugs). Now that I look at it, it looks like a good idea to remove the return value.

**Comment by BevapDin** [comment]
That fixes the item duplication, but there is still another issue:
Get a loaded flashlight, put it on the ground, open AIM (one pane shows the ground with the flashlight, the other shows the inventory), select the ground tile and enter the "Move all items" command.

Instead of one "flashlight (off) (100)", you'll pick up the flashlight charge by charges, but every time you get a flashlight for free. You'll end up with 100 x "flashlight (off) (1)".

Also works with ground-to-ground transfer (using the "Move all items" command).

**Pull request merged**